classes/Validate.php

```diff
@@ -360,11 +360,16 @@ public static function isDiscountName($voucher)
      *
      * @param string $name Product or category name to validate
      *
+     * product name has max length 128 in db
+     * category name has max length 128 in db
+     *
+     * this should really be handled in the DAL layer for each object and it should throw an error which can then be used elsewhere
+     *
      * @return bool Validity is ok or not
      */
     public static function isCatalogName($name)
     {
-        return preg_match(Tools::cleanNonUnicodeSupport('/^[^<>;=#{}]*$/u'), $name);
+        return Tools::strlen($name) <= 128;
```

Comment:
I'm never comfortable with value in code, especially since is duplicated in many place. Could you create a const that you will be able to re-use anywhere you want ?